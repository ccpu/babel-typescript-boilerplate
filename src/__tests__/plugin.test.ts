import path from 'path';
import glob from 'fast-glob';
import { transform } from './utils';

describe('should pass fixtures', () => {
  const files = glob.sync('./fixtures/**/*.{ts,tsx}', {
    cwd: __dirname,
    dot: false,
  });

  files.forEach((basePath) => {
    const filePath = String(basePath);

    if (filePath.includes('/special/') || filePath.includes('/typings/')) {
      return;
    }

    it(`transforms ${filePath}`, () => {
      expect(
        transform(path.join(__dirname, filePath), {
          comments: false,
          value: 'foo',
        }),
      ).toMatchSnapshot();
    });
  });
});
