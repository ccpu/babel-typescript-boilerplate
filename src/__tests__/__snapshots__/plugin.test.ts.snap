// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass fixtures transforms ./fixtures/interface-schema.ts 1`] = `"export const testVariable = 'foo';"`;
